docs/fundamentals/networking/httpclient.md

```diff
@@ -0,0 +1,293 @@
+
+#### HTTP Put as JSON
+
+To automatically serialize `PUT` request arguments and deserialize responses into strongly-typed C# objects, use the <xref:System.Net.Http.Json.HttpClientJsonExtensions.PutAsJsonAsync%2A> extension method that's part of the [System.Net.Http.Json](https://www.nuget.org/packages/System.Net.Http.Json) NuGet package.
+
+:::code language="csharp" source="snippets/httpclient/Program.PutAsJson.cs" id="putasjson":::
+
+The preceding code:
+
+- Serializes the `Todo` instance as JSON, and makes a `PUT` request to `"https://jsonplaceholder.typicode.com/todos/5"`.
+- Ensures that the response is successful, and writes the request details to the console.
+- Deserializes the response body into a `Todo` instance, and writes the `Todo` to the console.
+
+### HTTP Patch
+
+The `PATCH` request is a partial update to an existing resource. It will not create a new resource, nor is it intended to replace an existing resource, instead it updates a resource only partially. To make an HTTP `PATCH` request, given an `HttpClient` and a URI, use the <xref:System.Net.Http.HttpClient.PatchAsync%2A?displayProperty=nameWithType> method:
+
+:::code language="csharp" source="snippets/httpclient/Program.Patch.cs" id="patch":::
+
+The preceding code:
+
+- Prepares a <xref:System.Net.Http.StringContent> instance with the JSON body of the request (MIME type of `"application/json"`).
+- Makes a `PATCH` request to `"https://jsonplaceholder.typicode.com/todos/1"`.
+- Ensures that the response is successful, and writes the request details and JSON response body to the console.
+
+No extension methods exist for `PATCH` requests in the `System.Net.Http.Json` NuGet package.
+
+### HTTP Delete
+
+The `DELETE` request will delete an existing resource. A `DELETE` request is _idempotent_ but not _safe_, meaning multiple `DELETE` requests to the same resources yield the same result, but the request will affect the state of the resource. To make an HTTP `DELETE` request, given an `HttpClient` and a URI, use the <xref:System.Net.Http.HttpClient.DeleteAsync%2A?displayProperty=nameWithType> method:
+
+:::code language="csharp" source="snippets/httpclient/Program.Delete.cs" id="delete":::
+
+The preceding code:
+
+- Makes a `DELETE` request to `"https://jsonplaceholder.typicode.com/todos/1"`.
+- Ensures that the response is successful, and writes the request details to the console.
+
+> [!TIP]
+> The response to a `DELETE` request (just like a `PUT` request) may or may not include a body.
+
+### HTTP Head
+
+The `HEAD` request is similar to a `GET` request, but instead of returning the resource it only returns the headers associated with the resource. A response to the `HEAD` request will not return a body. To make an HTTP `HEAD` request, given an `HttpClient` and a URI, use the <xref:System.Net.Http.HttpClient.SendAsync%2A?displayProperty=nameWithType> method with the <xref:System.Net.Http.HttpMethod> set to `HttpMethod.Head`:
```

Comment:
```suggestion
The `HEAD` request is similar to a `GET` request. Instead of returning the resource, it only returns the headers associated with the resource. A response to the `HEAD` request doesn't return a body. To make an HTTP `HEAD` request, given an `HttpClient` and a URI, use the <xref:System.Net.Http.HttpClient.SendAsync%2A?displayProperty=nameWithType> method with the <xref:System.Net.Http.HttpMethod> set to `HttpMethod.Head`:
```